laborator/lab-05/README.md

```diff
@@ -0,0 +1,313 @@
+# Laborator 05: Rolul registrelor, adresare directă și bazată
+
+În acest laborator vom aprofunda lucrul cu registre și modul în care se utilizează memoria atunci când programăm assembly pe un sistem x86 de 32 biți.
+
+## Registre
+
+Registrele sunt principalele “unelte” cu care se scriu programele în limbaj de asamblare. Acestea sunt precum variabile construite în procesor. Utilizarea registrelor în locul adresării directe a memoriei face ca dezvoltarea și citirea programelor scrise în assembly să fie mai rapidă și mai ușoara. Singurul dezavantaj al programării în limbaj de asamblare x86 este acela că sunt puține registre.
+
+Procesoarele x86 moderne dispun de 8 registre cu scop general a căror dimensiune este de 32 de biți. Numele registrelor sunt de natură istorică (spre exemplu: EAX era numit registru acumulator din cauza faptului că este folosit de o serie de instrucțiuni aritmetice, cum ar fi [idiv](https://www.felixcloutier.com/x86/idiv)). In timp ce majoritatea registrelor și-au pierdut scopul special, devenind “general purpose” în ISA-ul modern(EAX, EBX, ECX, EDX, ESI, EDI), prin convenție, 2 și-au pastrat scopul inițial: ESP (stack pointer) și EBP (base pointer).
+
+### Subsecțiuni ale registrelor
+
+În anumite cazuri dorim să modificăm valori ce sunt reprezentate pe mai puțin de 4 octeți (spre exemplu, lucrul cu șiruri de caractere). Pentru aceste situații, procesoarele x86 ne oferă posibilitatea de a lucra cu subsectiuni de 1, respectiv 2 octeți ale registrelor EAX, EBX, ECX, EDX.
+
+În imaginea de mai jos sunt reprezentate registrele, subregistrele și dimensiunile lor.
+
+![Registre x86_32](https://www.tenouk.com/Bufferoverflowc/Bufferoverflow1_files/image005.png)
+
+> **WARNING**
+> Subregistrele fac parte din registre, ceea ce înseamnă că dacă modificăm un registru, în mod implicit modificăm și valoarea subregistrului.
```

Comment:
Adaugă tipul de notă pe aceeași linie cu restul textului, chiar dacă în fișierul final sunt generate ca fiind pe aceeași linie.
`**WARNING**: Subregistrele...`

Reply:
Fixed